# Table of Contents
 <p><div class="lev1 toc-item"><a href="#Exercises" data-toc-modified-id="Exercises-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Exercises</a></div><div class="lev2 toc-item"><a href="#1.-Consider-Equation-(16.3).-In-each-of-the-entries-in-the-table-below,-fill-in-either-R-for-random-or-NR-for-nonrandom:" data-toc-modified-id="1.-Consider-Equation-(16.3).-In-each-of-the-entries-in-the-table-below,-fill-in-either-R-for-random-or-NR-for-nonrandom:-11"><span class="toc-item-num">1.1&nbsp;&nbsp;</span>1. Consider Equation (16.3). In each of the entries in the table below, fill in either R for random or NR for nonrandom:</a></div><div class="lev2 toc-item"><a href="#2.-Consider-hat{p},-the-estimator-of-a-population-proportion-p,-based-on-a-sample-size-n.-Give-the-expression-for-the-standard-error-of-\hat{p}." data-toc-modified-id="2.-Consider-hat{p},-the-estimator-of-a-population-proportion-p,-based-on-a-sample-size-n.-Give-the-expression-for-the-standard-error-of-\hat{p}.-12"><span class="toc-item-num">1.2&nbsp;&nbsp;</span>2. Consider $\hat{p}$, the estimator of a population proportion $p$, based on a sample size $n$. Give the expression for the standard error of $\hat{p}$.</a></div><div class="lev2 toc-item"><a href="#3.-Suppose-we-take-a-simple-random-sample-of-size-2-from-a-population-consisting-of-just-three-values,-66,-67,-and-69.-Let-overline{X}-denote-the-resulting-sample-mean.-Find-p_{\overline{X}}(67.5)." data-toc-modified-id="3.-Suppose-we-take-a-simple-random-sample-of-size-2-from-a-population-consisting-of-just-three-values,-66,-67,-and-69.-Let-overline{X}-denote-the-resulting-sample-mean.-Find-p_{\overline{X}}(67.5).-13"><span class="toc-item-num">1.3&nbsp;&nbsp;</span>3. Suppose we take a simple random sample of size 2 from a population consisting of just three values, 66, 67, and 69. Let $\overline{X}$ denote the resulting sample mean. Find $p_{\overline{X}}(67.5)$.</a></div><div class="lev2 toc-item"><a href="#4.-Suppose-we-have-a-random-sample-W_1,-...-,-W_n-and-we-wish-to-estimate-the-population-mean-mu,-as-usual.-But-we-decide-to-place-double-weight-on-W_1,-so-our-estimator-for-\mu-is" data-toc-modified-id="4.-Suppose-we-have-a-random-sample-W_1,-...-,-W_n-and-we-wish-to-estimate-the-population-mean-mu,-as-usual.-But-we-decide-to-place-double-weight-on-W_1,-so-our-estimator-for-\mu-is-14"><span class="toc-item-num">1.4&nbsp;&nbsp;</span>4. Suppose we have a random sample $W_1, ... , W_n$ and we wish to estimate the population mean $\mu$, as usual. But we decide to place double weight on $W_1$, so our estimator for $\mu$ is</a></div><div class="lev2 toc-item"><a href="#5.-Suppose-a-random-sample-of-size-n-is-drawn-from-a-population-in-which,-unknown-to-the-analyst,-X-actually-has-an-exponential-distribution-with-mean-10.-Suppose-the-analyst-forms-an-approximate-95%-confidence-interval-for-the-mean,-using-(16.4).-Use-R-simulation-to-estimate-the-true-confidence-level,-for-n=10,25,100,-and-500." data-toc-modified-id="5.-Suppose-a-random-sample-of-size-n-is-drawn-from-a-population-in-which,-unknown-to-the-analyst,-X-actually-has-an-exponential-distribution-with-mean-10.-Suppose-the-analyst-forms-an-approximate-95%-confidence-interval-for-the-mean,-using-(16.4).-Use-R-simulation-to-estimate-the-true-confidence-level,-for-n=10,25,100,-and-500.-15"><span class="toc-item-num">1.5&nbsp;&nbsp;</span>5. Suppose a random sample of size $n$ is drawn from a population in which, unknown to the analyst, $X$ actually has an exponential distribution with mean $10$. Suppose the analyst forms an approximate $95%$ confidence interval for the mean, using (16.4). Use R simulation to estimate the true confidence level, for $n=10,25,100$, and $500$.</a></div><div class="lev2 toc-item"><a href="#6.-Suppose-we-draw-a-sample-of-size-2-from-a-population-in-which-X-has-the-values-10,-15,-and-12.-Find-p_{overline{X}},-first-assuming-sampling-with-replacement,-then-assuming-sampling-without-replacement." data-toc-modified-id="6.-Suppose-we-draw-a-sample-of-size-2-from-a-population-in-which-X-has-the-values-10,-15,-and-12.-Find-p_{overline{X}},-first-assuming-sampling-with-replacement,-then-assuming-sampling-without-replacement.-16"><span class="toc-item-num">1.6&nbsp;&nbsp;</span>6. Suppose we draw a sample of size $2$ from a population in which $X$ has the values $10, 15,$ and $12$. Find $p_{\overline{X}}$, first assuming sampling with replacement, then assuming sampling without replacement.</a></div><div class="lev2 toc-item"><a href="#7.-We-ask-100-randomly-sampled-programmers-whether-C++-is-their-favorite-language,-and-12-answer-yes.-Give-a-numerical-expression-for-an-approximate-95%-confidence-interval-for-the-population-fraction-of-programmers-who-have-C++-as-their-favorite-language." data-toc-modified-id="7.-We-ask-100-randomly-sampled-programmers-whether-C++-is-their-favorite-language,-and-12-answer-yes.-Give-a-numerical-expression-for-an-approximate-95%-confidence-interval-for-the-population-fraction-of-programmers-who-have-C++-as-their-favorite-language.-17"><span class="toc-item-num">1.7&nbsp;&nbsp;</span>7. We ask $100$ randomly sampled programmers whether C++ is their favorite language, and 12 answer yes. Give a numerical expression for an approximate $95%$ confidence interval for the population fraction of programmers who have C++ as their favorite language.</a></div><div class="lev2 toc-item"><a href="#8.-In-Equation-(16.4),-suppose-1.96-is-replaced-by-1.88-in-both-instances.-Then-of-course-the-confidence-level-will-be-smaller-than-95%.-Give-a-call-to-an-R-function-(not-a-simulation),-that-will-find-the-new-confidence-level." data-toc-modified-id="8.-In-Equation-(16.4),-suppose-1.96-is-replaced-by-1.88-in-both-instances.-Then-of-course-the-confidence-level-will-be-smaller-than-95%.-Give-a-call-to-an-R-function-(not-a-simulation),-that-will-find-the-new-confidence-level.-18"><span class="toc-item-num">1.8&nbsp;&nbsp;</span>8. In Equation (16.4), suppose $1.96$ is replaced by $1.88$ in both instances. Then of course the confidence level will be smaller than $95\%$. Give a call to an R function (not a simulation), that will find the new confidence level.</a></div><div class="lev2 toc-item"><a href="#9.-Candidates-A,-B-and-C-are-vying-for-election.-Let-p_1,-p_2-and-p_3-denote-the-fractions-of-people-planning-to-vote-for-them.-We-poll-n-people-at-random,-yielding-estimates-hat{p_1},-\hat{p_2},-and-\hat{p_3}.-Y-claims-that-she-has-more-supporters-than-the-other-two-candidates-combined.-Give-a-formula-for-an-approximate-95\%-confidence-interval-for-p_2-(p_1+p_3)." data-toc-modified-id="9.-Candidates-A,-B-and-C-are-vying-for-election.-Let-p_1,-p_2-and-p_3-denote-the-fractions-of-people-planning-to-vote-for-them.-We-poll-n-people-at-random,-yielding-estimates-hat{p_1},-\hat{p_2},-and-\hat{p_3}.-Y-claims-that-she-has-more-supporters-than-the-other-two-candidates-combined.-Give-a-formula-for-an-approximate-95\%-confidence-interval-for-p_2-(p_1+p_3).-19"><span class="toc-item-num">1.9&nbsp;&nbsp;</span>9. Candidates A, B and C are vying for election. Let $p_1, p_2$ and $p_3$ denote the fractions of people planning to vote for them. We poll $n$ people at random, yielding estimates $\hat{p_1}, \hat{p_2}$, and $\hat{p_3}$. $Y$ claims that she has more supporters than the other two candidates combined. Give a formula for an approximate $95\%$ confidence interval for $p_2-(p_1+p_3)$.</a></div><div class="lev2 toc-item"><a href="#10.-Suppose-Jack-and-Jill-each-collect-random-samples-of-size-n-from-a-population-having-unknown-mean-mu-but-KNOWN-variance-\sigma^2.-They-each-form-an-approximate-95\%-confidence-interval-for-\mu,-using-(16.4)-but-with-s-replaced-by-\sigma.-Find-the-approximate-probability-that-their-intervals-do-not-overlap.-Express-your-answer-in-terms-of-\phi,-the-cdf-of-the-N(0,1)-distribution." data-toc-modified-id="10.-Suppose-Jack-and-Jill-each-collect-random-samples-of-size-n-from-a-population-having-unknown-mean-mu-but-KNOWN-variance-\sigma^2.-They-each-form-an-approximate-95\%-confidence-interval-for-\mu,-using-(16.4)-but-with-s-replaced-by-\sigma.-Find-the-approximate-probability-that-their-intervals-do-not-overlap.-Express-your-answer-in-terms-of-\phi,-the-cdf-of-the-N(0,1)-distribution.-110"><span class="toc-item-num">1.10&nbsp;&nbsp;</span>10. Suppose Jack and Jill each collect random samples of size $n$ from a population having unknown mean $\mu$ but KNOWN variance $\sigma^2$. They each form an approximate $95\%$ confidence interval for $\mu$, using (16.4) but with $s$ replaced by $\sigma$. Find the approximate probability that their intervals do not overlap. Express your answer in terms of $\phi$, the cdf of the $N(0,1)$ distribution.</a></div><div class="lev2 toc-item"><a href="#11.-In-the-example-of-the-population-of-three-people,-page-263,-find-the-following:" data-toc-modified-id="11.-In-the-example-of-the-population-of-three-people,-page-263,-find-the-following:-111"><span class="toc-item-num">1.11&nbsp;&nbsp;</span>11. In the example of the population of three people, page 263, find the following:</a></div><div class="lev3 toc-item"><a href="#(a)-p_{X_1}(70)" data-toc-modified-id="(a)-p_{X_1}(70)-1111"><span class="toc-item-num">1.11.1&nbsp;&nbsp;</span>(a) $p_{X_1}(70)$</a></div><div class="lev3 toc-item"><a href="#(b)-p_{X_1,X_2}(69,70)" data-toc-modified-id="(b)-p_{X_1,X_2}(69,70)-1112"><span class="toc-item-num">1.11.2&nbsp;&nbsp;</span>(b) $p_{X_1,X_2}(69,70)$</a></div><div class="lev3 toc-item"><a href="#(c)-F_{overline{X}}(69.5)" data-toc-modified-id="(c)-F_{overline{X}}(69.5)-1113"><span class="toc-item-num">1.11.3&nbsp;&nbsp;</span>(c) $F_{\overline{X}}(69.5)$</a></div><div class="lev3 toc-item"><a href="#(d)-probability-that-overline{X}-overestimates-the-population-mean-\mu" data-toc-modified-id="(d)-probability-that-overline{X}-overestimates-the-population-mean-\mu-1114"><span class="toc-item-num">1.11.4&nbsp;&nbsp;</span>(d) probability that $\overline{X}$ overestimates the population mean $\mu$</a></div><div class="lev3 toc-item"><a href="#(e)-p_{overline{X}}(69)-if-our-sample-size-is-three-rather-than-two-(remember,-we-are-sampling-with-replacement)" data-toc-modified-id="(e)-p_{overline{X}}(69)-if-our-sample-size-is-three-rather-than-two-(remember,-we-are-sampling-with-replacement)-1115"><span class="toc-item-num">1.11.5&nbsp;&nbsp;</span>(e) $p_{\overline{X}}(69)$ if our sample size is three rather than two (remember, we are sampling with replacement)</a></div><div class="lev2 toc-item"><a href="#12.-In-the-derivation-(15.8),-suppose-instead-we-have-a-simple-random-sample.-Which-one-of-the-following-statements-is-correct?" data-toc-modified-id="12.-In-the-derivation-(15.8),-suppose-instead-we-have-a-simple-random-sample.-Which-one-of-the-following-statements-is-correct?-112"><span class="toc-item-num">1.12&nbsp;&nbsp;</span>12. In the derivation (15.8), suppose instead we have a simple random sample. Which one of the following statements is correct?</a></div><div class="lev3 toc-item"><a href="#(a)-E(overline{X})-will-still-be-equal-to-\mu." data-toc-modified-id="(a)-E(overline{X})-will-still-be-equal-to-\mu.-1121"><span class="toc-item-num">1.12.1&nbsp;&nbsp;</span>(a) $E(\overline{X})$ will still be equal to $\mu$.</a></div><div class="lev3 toc-item"><a href="#(b)-E(overline{X})-will-not-exist." data-toc-modified-id="(b)-E(overline{X})-will-not-exist.-1122"><span class="toc-item-num">1.12.2&nbsp;&nbsp;</span>(b) $E(\overline{X})$ will not exist.</a></div><div class="lev3 toc-item"><a href="#(c)-E(overline{X})-will-exist,-but-may-be-less-than-\mu." data-toc-modified-id="(c)-E(overline{X})-will-exist,-but-may-be-less-than-\mu.-1123"><span class="toc-item-num">1.12.3&nbsp;&nbsp;</span>(c) $E(\overline{X})$ will exist, but may be less than $\mu$.</a></div><div class="lev3 toc-item"><a href="#(d)-E(overline{X})-will-exist,-but-may-be-greater-than-\mu." data-toc-modified-id="(d)-E(overline{X})-will-exist,-but-may-be-greater-than-\mu.-1124"><span class="toc-item-num">1.12.4&nbsp;&nbsp;</span>(d) $E(\overline{X})$ will exist, but may be greater than $\mu$.</a></div><div class="lev3 toc-item"><a href="#(e)-None-of-the-above-is-necessarily-true." data-toc-modified-id="(e)-None-of-the-above-is-necessarily-true.-1125"><span class="toc-item-num">1.12.5&nbsp;&nbsp;</span>(e) None of the above is necessarily true.</a></div><div class="lev2 toc-item"><a href="#13.-Consider-a-toy-example-in-which-we-take-a-random-sample-of-size-2-(done-with-replacement)-from-a-population-of-size-2.-The-two-values-in-the-population-(say-heights-in-some-measure-system)-are-40-and-60.-Find-p_{s^2}(100)." data-toc-modified-id="13.-Consider-a-toy-example-in-which-we-take-a-random-sample-of-size-2-(done-with-replacement)-from-a-population-of-size-2.-The-two-values-in-the-population-(say-heights-in-some-measure-system)-are-40-and-60.-Find-p_{s^2}(100).-113"><span class="toc-item-num">1.13&nbsp;&nbsp;</span>13. Consider a toy example in which we take a random sample of size $2$ (done with replacement) from a population of size $2$. The two values in the population (say heights in some measure system) are $40$ and $60$. Find $p_{s^2}(100)$.</a></div>

In [77]:
import numpy as np
from scipy.stats import norm
from scipy.stats import multivariate_normal

# Exercises

## 1. Consider Equation (16.3). In each of the entries in the table below, fill in either R for random or NR for nonrandom:

-------------------
| quantity | R or NR?|
|----------|---------|
|$\overline{W}$| R |
| $s$ | R |
| $\mu$ | NR |
| $n$ | NR |
-----------

## 2. Consider $\hat{p}$, the estimator of a population proportion $p$, based on a sample size $n$. Give the expression for the standard error of $\hat{p}$.

Consider example of people voting for Candidate A. Let $Y_i = 0$ if person $i$ is not voting for A, and $1$ otherwise. 

$$
\begin{equation}
\begin{aligned}
    s^2 &= \sum_{i=1}^n (Y_i^2) - \overline{Y}^2 \\
    &= \sum_{i=1}^n Y_i - \overline{Y}^2 \\
    &= \hat{p} - \hat{p}^2 && \text{$\hat{p} = \overline{Y}$} \\
\end{aligned}
\end{equation}
$$

$$
\begin{equation}
\begin{aligned}
    s.e.(Y) &= \sqrt{\frac{s^2}{n}} \\
    &= \sqrt{\frac{\hat{p}-\hat{p}^2}{n}}
\end{aligned}
\end{equation}
$$

## 3. Suppose we take a simple random sample of size 2 from a population consisting of just three values, 66, 67, and 69. Let $\overline{X}$ denote the resulting sample mean. Find $p_{\overline{X}}(67.5)$.

Let $X_1$ be the first pick and $X_2$ be the second. We are interested in $p_{\overline{X}}(67.5) = P(\frac{X_1 + X_2}{2} = 67.5)$. Assuming random sampling with replacement, the only way to get this result is by picking a $66$ and a $69$, and vice versa.

$$
\begin{equation}
\begin{aligned}
    P\Big(\frac{X_1 + X_2}{2} = 67.5\Big) &= P(X_1=66, X_2=69) + P(X_1=69, X_2 = 66) \\
    &= P(X_1=66)P(X_2=69) + P(X_1=69)P(X_2 = 66) && \text{indep.} \\
    &= \frac{1}{3}\cdot\frac{1}{3}+\frac{1}{3}\cdot\frac{1}{3} \\
    &= \frac{1}{9}
\end{aligned}
\end{equation}
$$


## 4. Suppose we have a random sample $W_1, ... , W_n$ and we wish to estimate the population mean $\mu$, as usual. But we decide to place double weight on $W_1$, so our estimator for $\mu$ is

$$
\begin{equation}
\begin{aligned}
    U &= \frac{2W_1 + W_2 + ... + W_n}{n+1}
\end{aligned}
\end{equation}
$$

Find $E(U)$ and $Var(U)$ in terms of $\mu$ and the population variance $\sigma^2$.

$$
\begin{equation}
\begin{aligned}
    E(U) &= \frac{\sum_{i=1}^{n+1}W_i}{n+1} && \text{Let $W_{n+1}=W_1$} \\
    &= \overline{W} \\ 
    &= \mu
\end{aligned}
\end{equation}
$$


$$
\begin{equation}
\begin{aligned}
    Var(U) &= Var\Big(\frac{2W_1+W_2+...+W_n}{n+1}\Big) \\
    &= Var\Big(\frac{2}{n+1}\cdot W_1\Big) + ... + Var\Big(\frac{W_n}{n+1}\Big) \\
    &= \frac{4}{(n+1)^2}\cdot \sigma^2 + \frac{n-1}{(n+1)^2}\cdot \sigma^2 \\
    &= \frac{n+3}{(n+1)^2}\cdot \sigma^2
\end{aligned}
\end{equation}
$$

Double weighting $W_1$ seems to keep the mean the same as not double weighting it. However, the former slightly increases the variance (compared to $\frac{\sigma^2}{n}$).

## 5. Suppose a random sample of size $n$ is drawn from a population in which, unknown to the analyst, $X$ actually has an exponential distribution with mean $10$. Suppose the analyst forms an approximate $95%$ confidence interval for the mean, using (16.4). Use R simulation to estimate the true confidence level, for $n=10,25,100$, and $500$.

In [3]:
np.random.exponential(scale=10,size=10)

array([ 23.85602335,  10.64944248,   5.11911264,  30.42118786,
        16.04987466,   7.04727479,  12.03693781,   3.19927576,
         1.45681344,  16.22023303])

In [13]:
norm.ppf(0.95)

1.6448536269514722

In [16]:
norm.ppf(0.025)

-1.9599639845400545

In [18]:
def ci(samples):
    mean = samples.mean()
    n = len(samples)
    var = samples.var()
    
    return (
        mean-1.96*(var/n)**0.5,
        mean+1.96*(var/n)**0.5
    )

In [39]:
ci(np.random.exponential(scale=10,size=10))

(4.641392927077594, 19.828153110949426)

In [27]:
ci(np.random.exponential(scale=10,size=25))

(6.5423782450710473, 17.816808745603467)

In [28]:
ci(np.random.exponential(scale=10,size=100))

(9.2056812568392843, 13.482012612096408)

In [30]:
ci(np.random.exponential(scale=10,size=500))

(9.3671140666133645, 11.022289752088664)

In [31]:
ci(np.random.exponential(scale=10,size=1000))

(9.3115490697973442, 10.630375739083842)

In [32]:
ci(np.random.exponential(scale=10,size=10000))

(9.9473462258815957, 10.337792012346807)

In [38]:
ci(np.random.exponential(scale=10,size=100000))

(9.8806994628912257, 10.003870074322418)

## 6. Suppose we draw a sample of size $2$ from a population in which $X$ has the values $10, 15,$ and $12$. Find $p_{\overline{X}}$, first assuming sampling with replacement, then assuming sampling without replacement.

Let $X_1$ be the first pick, and $X_2$ be the second. With replacement, here are all the possible outcomes:

-------------
|$X_1$ | $X_2$ | $\overline{X}$ |
|------|-------|---------|
|10|10|10|
|10|15|12.5|
|10|12|11|
|12|10|11|
|12|12|12|
|12|15|13.5|
|15|10|12.5|
|15|12|13.5|
|15|15|15|
----------

Thus $p_{\overline{X}}$ is the following:

-------
|   event   |probability|
|------------|-----------|
|$\overline{X}=10$|$\frac{1}{9}$|
|$\overline{X}=15$|$\frac{1}{9}$|
|$\overline{X}=12$|$\frac{1}{9}$|
|$\overline{X}=11$|$\frac{2}{9}$|
|$\overline{X}=12.5$|$\frac{2}{9}$|
|$\overline{X}=13.5$|$\frac{2}{9}$|

Without replacement, distribution of $\overline{X}$ becomes the following:

-------
|   event   |probability|
|------------|-----------|
|$\overline{X}=11$|$\frac{1}{3}$|
|$\overline{X}=12.5$|$\frac{1}{3}$|
|$\overline{X}=13.5$|$\frac{1}{3}$|

## 7. We ask $100$ randomly sampled programmers whether C++ is their favorite language, and 12 answer yes. Give a numerical expression for an approximate $95%$ confidence interval for the population fraction of programmers who have C++ as their favorite language.

$\frac{12}{100} \pm 1.96\sqrt{\frac{
    \frac{12}{100}\Big(1 - \frac{12}{100}\Big)
}{100}}$

In [40]:
12/100.0 - 1.96*((12/100.0*(1-12/100.0))/100)**0.5

0.05630753890765407

In [41]:
12/100.0 + 1.96*((12/100.0*(1-12/100.0))/100)**0.5

0.18369246109234594

In [53]:
# Another way
norm.interval(0.95, loc=0.12, scale=np.sqrt(0.12*0.88 / 100))

(0.05630870927157311, 0.18369129072842688)

## 8. In Equation (16.4), suppose $1.96$ is replaced by $1.88$ in both instances. Then of course the confidence level will be smaller than $95\%$. Give a call to an R function (not a simulation), that will find the new confidence level.

In [47]:
norm.ppf(0.03)
# Thus the CI would be 1 - 2*0.03 = 94%

-1.8807936081512511

In [52]:
# Applying to question 7, changing 0.95 to 0.94
norm.interval(0.94, loc=0.12, scale=np.sqrt(0.12*0.88 / 100))

(0.0588814419847416, 0.18111855801525839)

## 9. Candidates A, B and C are vying for election. Let $p_1, p_2$ and $p_3$ denote the fractions of people planning to vote for them. We poll $n$ people at random, yielding estimates $\hat{p_1}, \hat{p_2}$, and $\hat{p_3}$. $Y$ claims that she has more supporters than the other two candidates combined. Give a formula for an approximate $95\%$ confidence interval for $p_2-(p_1+p_3)$.

Let $A_i$ be $1$ if person $i$ votes for Candidate $A$, $0$ otherwise, etc.


$$
\begin{equation}
\begin{aligned}
    Var(\hat{p_2}-(\hat{p_1} + \hat{p_3})) &= Var(\hat{p_2}) + Var(\hat{p_1}) + Var(\hat{p_3}) \\
    &\quad - 2Cov(\hat{p_2}, \hat{p_1}) + 2Cov(\hat{p_3}, \hat{p_1}) - 2Cov(\hat{p_2}, \hat{p_1}) \\
    &= \frac{1}{n}\sum_{i=0}^n A_i^2 - \overline{A}^2 \\
    &\quad + \frac{1}{n}\sum_{i=0}^n B_i^2 - \overline{B}^2 \\
    &\quad + \frac{1}{n}\sum_{i=0}^n C_i^2 - \overline{B}^2 \\
    &\quad -2\Big(\frac{1}{n}\sum_{i=0}^n A_iB_i - (\overline{A}\overline{B})^2\Big) \\
    &\quad -2\Big(\frac{1}{n}\sum_{i=0}^n B_iC_i - (\overline{B}\overline{C})^2\Big) \\
    &\quad +2\Big(\frac{1}{n}\sum_{i=0}^n A_iC_i - (\overline{A}\overline{C})^2\Big) \\
    &= \frac{1}{n}\sum_{i=0}^n A_i - \overline{A}^2 && \text{$A_i^2=A_i$ since $A_i$ is $0$ or $1$}\\
    &\quad + \frac{1}{n}\sum_{i=0}^n B_i - \overline{B}^2 \\
    &\quad + \frac{1}{n}\sum_{i=0}^n C_i - \overline{B}^2 \\
    &\quad -2\Big(\frac{1}{n}\sum_{i=0}^n A_iB_i - \overline{A}\overline{B}\Big) \\
    &\quad -2\Big(\frac{1}{n}\sum_{i=0}^n B_iC_i - \overline{B}\overline{C}\Big) \\
    &\quad +2\Big(\frac{1}{n}\sum_{i=0}^n A_iC_i - \overline{A}\overline{C}\Big) \\
    &= \hat{p_1}(1-\hat{p_1}) + \hat{p_2}(1-\hat{p_2}) + \hat{p_3}(1-\hat{p_3}) \\
    &\quad -2\Big(\frac{1}{n}\sum_{i=0}^n A_iB_i - \overline{A}\overline{B}\Big) \\
    &\quad -2\Big(\frac{1}{n}\sum_{i=0}^n B_iC_i - \overline{B}\overline{C}\Big) \\
    &\quad +2\Big(\frac{1}{n}\sum_{i=0}^n A_iC_i - \overline{A}\overline{C}\Big) \\
    &= \hat{p_1}(1-\hat{p_1}) + \hat{p_2}(1-\hat{p_2}) + \hat{p_3}(1-\hat{p_3}) \\
    &\quad + 2\hat{p_1}\hat{p_2} + 2\hat{p_2}\hat{p_3} - 2\hat{p_1}\hat{p_3} \\
\end{aligned}
\end{equation}
$$


Thus, the $95\%$ CI for $\hat{p_2}-(\hat{p_1}-\hat{p_3})$ is $\hat{p_2}-(\hat{p_1}-\hat{p_3}) \pm 1.96 \cdot \sqrt{\frac{\hat{p_1}(1-\hat{p_1}) + \hat{p_2}(1-\hat{p_2}) + \hat{p_3}(1-\hat{p_3}) + 2\hat{p_1}\hat{p_2} + 2\hat{p_2}\hat{p_3} - 2\hat{p_1}\hat{p_3}}{n}}$


## 10. Suppose Jack and Jill each collect random samples of size $n$ from a population having unknown mean $\mu$ but KNOWN variance $\sigma^2$. They each form an approximate $95\%$ confidence interval for $\mu$, using (16.4) but with $s$ replaced by $\sigma$. Find the approximate probability that their intervals do not overlap. Express your answer in terms of $\phi$, the cdf of the $N(0,1)$ distribution.

In [55]:
1-norm.cdf(0.894)

0.18566094557883606

In [56]:
norm.cdf(-1.96)

0.024997895148220435

In [57]:
norm.cdf(1.96)

0.97500210485177952

In [163]:
norm.interval(0.99,loc=0,scale=1)

(-2.5758293035489004, 2.5758293035489004)

In [150]:
def sim_jack_jill(
    num_samples = 100,
    num_experiments = 10000,
    known_variance = 1
):
    
    non_overlap_count = 0
    jill_means = []
    jack_means = []
    
    for i in range(num_experiments):
        
        jack_samples = np.random.normal(
            size=num_samples,
            scale=known_variance
        )
        
        jill_samples = np.random.normal(
            size=num_samples,
            scale=known_variance
        )
        
        jill_mean = jill_samples.mean()
        jack_mean = jack_samples.mean()
        
        jill_means.append(jill_mean)
        jack_means.append(jack_mean)
        
        jack_lower_bound, jack_upper_bound = norm.interval(
            0.95, 
            loc=jack_mean, 
            scale=np.sqrt(known_variance / num_samples)
        )
        
        jill_lower_bound, jill_upper_bound = norm.interval(
            0.95, 
            loc=jill_mean, 
            scale=np.sqrt(known_variance / num_samples)
        )
        
        if (jack_upper_bound < jill_lower_bound 
            or jill_upper_bound < jack_lower_bound):
            non_overlap_count += 1
            
        
    
    return (non_overlap_count / float(num_experiments), 
            jill_means, 
            jack_means,
            known_variance,
            num_samples
           )
        
        

In [123]:
non_overlap_mean, jill_means, jack_means = sim_jack_jill()

In [ ]:
1

In [124]:
non_overlap_mean

0.0065

In [179]:
2-2*norm.cdf(3.92/np.sqrt(2))

0.005573724535172131

In [182]:
1-2*norm.cdf(3.92/np.sqrt(2))

-0.99442627546482787

In [181]:
vals = sim_jack_jill(
    known_variance=2
)
vals[0]

0.0531

In [126]:
print('Jill means: ', jill_means[0], ' Jack means: ', jack_means[0])

Jill means:  -0.102857173591  Jack means:  -0.0316004795441


In [177]:
(jill_means[0] - jack_means[0])/np.sqrt(1.0/100.0)

-0.71256694046797331

In [178]:
(jill_means[0] - jack_means[0])/np.sqrt(2*1.0/100.0)

-0.50386091565425495

In [129]:
jill_means[0] - jack_means[0] - 1.96*np.sqrt(2/100)

-0.348442552271924

In [130]:
jill_means[0] - jack_means[0] + 1.96*np.sqrt(2/100)

0.20592916417832929

In [136]:
(jill_means[0] - jack_means[0])/np.sqrt(2/100)

-0.50386091565425495

In [142]:
norm.cdf((jill_means[0] - jack_means[0])/np.sqrt(1/100) - 1.96)*2

0.0075273354534704118

In [172]:
def approx_proba_jill_jack_overlap(
    results
):
    sim_proba = results[0]
    jill_mean = results[1]
    jack_mean = results[2]
    known_variance = results[3]
    num_samples = results[4]
    
    estim_proba = norm.cdf(
        (jill_mean[0] - jack_mean[0])/np.sqrt(
            2*known_variance/float(num_samples)
        ) - 1.96
    ) 
    
#     + 1 - norm.cdf(
#         (jill_mean[0] - jack_mean[0])/np.sqrt(
#             2*known_variance/float(num_samples)
#         ) + 1.96
#     )
    
    return sim_proba, estim_proba

In [173]:
approx_proba_jill_jack_overlap(
    sim_jack_jill(
        num_samples = 1000,
        num_experiments = 10000,
        known_variance = 1
    )
)

(0.0053, 0.080610300838290128)

In [174]:
approx_proba_jill_jack_overlap(
    sim_jack_jill(
        num_samples = 10,
        num_experiments = 10000,
        known_variance = 1
    )
)

(0.004, 0.22444622265684105)

In [175]:
approx_proba_jill_jack_overlap(
    sim_jack_jill(
        num_samples = 100,
        num_experiments = 10000,
        known_variance = 2
    )
)

(0.0461, 0.0038102523984158399)

In [155]:
1 - norm.cdf((jill_means[0] - jack_means[0])/np.sqrt(1/100) + 1.96)

0.10611937547730632

In [132]:
norm.cdf(-0.3484)

0.36376990126223635

In [134]:
1- norm.cdf(.2059)

0.41843451272422039

In [127]:
norm.cdf(-0.102857-1.96*1/10.0)

0.38252457873727808

In [113]:
sim_jack_jill(num_experiments=100000)

0.00614

In [114]:
sim_jack_jill(known_variance=2)

0.0509

In [115]:
sim_jack_jill(known_variance=2, num_experiments=100000)

0.05003

In [73]:
norm.cdf(0)**2

0.25

In [118]:
2*norm.cdf(-1.96*np.sqrt(2)/10)**2

0.30547851087010536

In [108]:
samps = np.random.multivariate_normal(
    mean=[0,0],
    cov=[
        [1,0 ],
        [0,1 ]
    ],
    size=100000
); samps

array([[ 0.24847129, -0.87107012],
       [-1.36702977,  1.93345941],
       [-1.50152161, -0.18800687],
       ..., 
       [-1.10665808, -0.57836826],
       [-0.93147905, -0.66316393],
       [ 0.17826474, -0.34825594]])

In [109]:
(
    (
        (samps.T[0] < -1.96) * (samps.T[1] > 1.96)
    ).sum() + (
        (samps.T[1] < -1.96) * (samps.T[0] > 1.96)
    ).sum()
) / float(100000)


0.0012099999999999999

## 11. In the example of the population of three people, page 263, find the following:

### (a) $p_{X_1}(70)$


### (b) $p_{X_1,X_2}(69,70)$

### (c) $F_{\overline{X}}(69.5)$


### (d) probability that $\overline{X}$ overestimates the population mean $\mu$

### (e) $p_{\overline{X}}(69)$ if our sample size is three rather than two (remember, we are sampling with replacement)

## 12. In the derivation (15.8), suppose instead we have a simple random sample. Which one of the following statements is correct?

### (a) $E(\overline{X})$ will still be equal to $\mu$.

### (b) $E(\overline{X})$ will not exist.

### (c) $E(\overline{X})$ will exist, but may be less than $\mu$.

### (d) $E(\overline{X})$ will exist, but may be greater than $\mu$.

### (e) None of the above is necessarily true.

## 13. Consider a toy example in which we take a random sample of size $2$ (done with replacement) from a population of size $2$. The two values in the population (say heights in some measure system) are $40$ and $60$. Find $p_{s^2}(100)$.